src/H5I.c

```diff
@@ -169,15 +179,25 @@ H5I_term_package(void)
 
         /* How many types are still being used? */
         for (type = 0; type < H5I_next_type; type++)
-            if ((type_ptr = H5I_id_type_list_g[type]) && type_ptr->ids)
-                n++;
+            if (H5I_IS_LIB_TYPE(type)) {
```

Comment:
Is there really a concern about the different type of pointers here?  Perhaps adding a void* to the union for when you want to operate on the data structure in a generic way?

Reply:
I thought about that, but was unconvinced a bunch of extra boilerplate was "better" than a half dozen if statements. It's not like we're going to expand this code to more ID data containers.